bindings/java/src/test/java/org/openucx/jucx/UcpListenerTest.java

```diff
@@ -126,6 +126,17 @@ public void testConnectionHandler() {
 
         UcpRequest sent = serverToClient.sendStreamNonBlocking(
             ByteBuffer.allocateDirect(UcpMemoryTest.MEM_SIZE), null);
+
+        // Progress all workers to make sure recv request will complete immediately
+        for (int i = 0; i < 10; i++) {
+            serverWorker1.progress();
+            serverWorker2.progress();
+            clientWorker.progress();
+            try {
+                Thread.sleep(2);
+            } catch (Exception ignored) { }
+        }
```

Comment:
do we have other test for non-immediate path?

Reply:
Yes: https://github.com/openucx/ucx/blob/master/bindings/java/src/test/java/org/openucx/jucx/UcpEndpointTest.java#L401